### PR TITLE
risc-v/mpfs: ihc: don't wait for master if it's already up

### PR DESCRIPTION
**Pull request by eenurkka**
After rebooting this ihc client hart only, ihc will hang as it's waiting for the initial handshake that just won't be there. Detect this situation and let the rptun / ihc know the master is already up. This makes the RPMSG bus usable even after NuttX reboot, in case there's not much traffic. However, Linux is likely to mark the virtio queue broken if there's descent amount of traffic. This makes the bus no longer function.

## Summary

Fix NuttX <-> Linux RPMSG after NuttX reboots

## Impact

NuttX <-> Linux RPMSG bus after NuttX reboot

## Testing

Saluki v2

### CODE REVIEW

**Comment by eenurkka** [comment]
Will send new version soon

**Review by jlaitine — APPROVED** [review]

File: arch/risc-v/src/mpfs/mpfs_ihc.c [hunk 1]
```diff
@@ -1236,7 +1239,20 @@ static void mpfs_rptun_worker(void *arg)
...
+  /* Check whether the struct is initialized yet */
+
+  if (*(uintptr_t *)&g_mpfs_virtqueue_table[0] == 0)
+    {
+      return;
+    }
+
+  /* g_vq_idx may be temporarily illegal after a warm boot */
+
+  if ((g_vq_idx - VRING0_NOTIFYID) >= VRINGS)
```
Comment (jlaitine) [inline arch/risc-v/src/mpfs/mpfs_ihc.c:1251]: Somehow I find this fishy.

After a warm re-boot, initially g_vq_idx is initialized to 0, whch == VRING0_NOTIFYID . Thus after a warm reboot this just always passes?

Would it be cleaner to just initialize g_vq_idx to some invalid value (e.g. 0xffff) and compare directly to that?

Comment (eenurkka) [reply arch/risc-v/src/mpfs/mpfs_ihc.c:1251]: After a warm boot, g_vq_idx is sometimes (1/10 or so) garbaged, so it won't be 0xffff, maybe the illegal value comes from the msg. Maybe I need to dig in more where the garbaged semi-random value comes from.

But setting it to an illegal value, eg 0xffff sounds like a good idea, but cannot compare to that as g_vq_idx is a rather random value when it corrupts.

Comment (eenurkka) [reply arch/risc-v/src/mpfs/mpfs_ihc.c:1251]:
```
 .sbss.g_rptun_initialized
                0x00000000a060f30c        0x1 ../../NuttX/nuttx/arch/risc-v/src/libarch.a(mpfs_ihc.o)
 *fill*         0x00000000a060f30d        0x1
 .sbss.g_vq_idx
                0x00000000a060f30e        0x2 ../../NuttX/nuttx/arch/risc-v/src/libarch.a(mpfs_ihc.o)
 .sbss._ZGVN10ModuleBaseI3ADCE7_objectE
                0x00000000a060f310        0x8 ../../src/drivers/adc/board_adc/libdrivers__board_adc.a(ADC.cpp.obj)
                0x00000000a060f310                guard variable for ModuleBase<ADC>::_object
 .sbss._ZN7cm8jl655g_devE
                0x00000000a060f318        0x8 ../../src/drivers/distance_sensor/cm8jl65/libdrivers__cm8jl65.a(cm8jl65_m>                0x00000000a060f318                cm8jl65::g_dev
```
I'll check if the neighbors overflow...

Comment (jlaitine) [reply arch/risc-v/src/mpfs/mpfs_ihc.c:1251]: I guess it can only come through  mpfs_ihc_interrupt, is there some spurious irq at boot up?

**Pull request merged**